File: tpc/java_version/postfix_infix.java
import datastructure.stack;
import java.util.*;

class postfix_infix{
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    String str = sc.nextLine();
    stack<String> s = new stack<>();

    for (char curr : str.toCharArray() ) {
      if (Character.isLetter(curr)) {
        s.push(String.valueOf(curr));
      }
      else{
        String a = s.pop();
        String b = s.pop();

        String temp = b+curr+a;
        s.push(temp);
      }

    }

    System.out.println(s.pop());

    sc.close();

  }
}

File: tpc/postfix_infix.py
import datastructure as ds


str = input()
s = ds.Stack()


for curr in str:
    if curr.isalpha():
        s.push(curr)
    else:
        a = s.pop()
        b = s.pop()
        temp = b + curr + a
        s.push(temp)

print(s.pop())
